src/main/java/org/apache/maven/plugins/toolchain/DisplayDiscoveredToolchainsMojo.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugins.toolchain;
+
+import java.io.StringWriter;
+
+import org.apache.maven.api.Session;
+import org.apache.maven.api.plugin.MojoException;
+import org.apache.maven.api.plugin.annotations.Component;
+import org.apache.maven.api.plugin.annotations.Mojo;
+import org.apache.maven.api.services.xml.ToolchainsXmlFactory;
+import org.apache.maven.api.toolchain.PersistedToolchains;
+import org.apache.maven.toolchain.discovery.ToolchainDiscoverer;
+
+@Mojo(name = "display-discovered-toolchains", requiresProject = false)
```

Comment:
With this name, I'd expect the goal to print a human-readable list of toolchains.

On the other hand, I think there's a lot of value in generating the **toolchains.xml** (as this Mojo currently does), but I'd expect that goal to be something like `generate-toolchains-xml`.

How would you feel about having two goals for those two use cases?